tests/test_server.py

```diff
@@ -323,6 +328,50 @@ def check_sc_override(converted):
         )
         self.assertEqual(dataset_type_component_data, MetricTestRepo.METRICS_EXAMPLE_SUMMARY)
 
+    def test_getURIs_no_components(self):
+        # This dataset does not have components, and should return one URI.
+        def check_uri(uri: ResourcePath):
+            self.assertIsNotNone(uris.primaryURI)
+            self.assertEqual(uris.primaryURI.scheme, "https")
+            self.assertEqual(uris.primaryURI.read(), b"123")
+
+        uris = self.butler.getURIs(self.simple_dataset_ref)
+        self.assertEqual(len(uris.componentURIs), 0)
+        check_uri(uris.primaryURI)
+
+        check_uri(self.butler.getURI(self.simple_dataset_ref))
+
+    def test_getURIs_multiple_components(self):
+        # This dataset has multiple components, so we should get back multiple
+        # URIs.
+        dataset_type = "test_metric_comp"
+        data_id = {"instrument": "DummyCamComp", "visit": 423}
+        collections = "ingest/run"
+
+        def check_uris(uris: DatasetRefURIs):
+            self.assertIsNone(uris.primaryURI)
+            self.assertEqual(len(uris.componentURIs), 3)
+            path = uris.componentURIs["summary"]
+            self.assertEqual(path.scheme, "https")
+            data = path.read()
+            self.assertEqual(data, b"AM1: 5.2\nAM2: 30.6\n")
+
+        uris = self.butler.getURIs(dataset_type, dataId=data_id, collections=collections)
+        check_uris(uris)
+
+        # Calling getURI on a multi-file dataset raises an exception
+        with self.assertRaises(RuntimeError):
+            self.butler.getURI(dataset_type, dataId=data_id, collections=collections)
+
+        # getURIs does NOT respect component overrides on the DatasetRef,
```

Comment:
There is only one URI for any of the components if not disassembled. I imagine no-one has ever tried to request the URI of an individual component by passing in the ref but it seems like it should return the one URI of the component.